### PR TITLE
chore: Allow  delombok

### PR DESCRIPTION
**Pull request by Nurkambay**
## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/test/java/org/eclipse/lsp/cobol/dialects/ibm/CobolLineIndicatorProcessorImplTest.java [hunk 1]
```diff
@@ -22,7 +22,6 @@
 import com.google.common.collect.ImmutableList;
 import java.util.List;
 import java.util.Optional;
-import lombok.val;
```
Comment (Nurkambay) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/dialects/ibm/CobolLineIndicatorProcessorImplTest.java:25]: This feature was not automatically delomboked, so I decided to remove it

File: delombok.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+java -jar lombok.jar delombok server -d server-delomboked
```
Comment (Nurkambay) [inline delombok.sh]: Run this file to delombok all sources to `server-delomboked` folder

File: server/engine/src/test/java/org/eclipse/lsp/cobol/dialects/ibm/CobolLineIndicatorProcessorImplTest.java (hunk 1 above)
Comment (slavek-kucera) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/dialects/ibm/CobolLineIndicatorProcessorImplTest.java:25]: Which version of Java is actually being targeted?
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/0d28f1bf5cc49cb956ad21e5fa8cd28650671c66/server/pom.xml#L23-L24
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/0d28f1bf5cc49cb956ad21e5fa8cd28650671c66/.github/workflows/build.yml#L90

Asking for a friend - https://openjdk.org/jeps/286

Comment (Nurkambay) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/dialects/ibm/CobolLineIndicatorProcessorImplTest.java:25]: Java 8, but you can use Jdk 11 to compile it

**Review by ishche — APPROVED** [review]

**Comment by slavek-kucera** [comment]
I am getting errors like
```
...server\common\src\main\java\org\eclipse\lsp\cobol\common\action\CodeActionProvider.java:19: error: package org.eclipse.lsp4j does not exist
import org.eclipse.lsp4j.CodeAction;
```
and eventually
```
...server\common\src\main\java\org\eclipse\lsp\cobol\common\mapping\ExtendedDocument.java:22: error: package org.eclipse.lsp4j does not exist
import org.eclipse.lsp4j.Position;
                        ^
Exception in thread "main" com.sun.tools.javac.processing.AnnotationProcessingError: java.lang.StackOverflowError
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1033)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:937)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1265)
        at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1380)
        at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1279)
        at lombok.delombok.Delombok.delombok(Delombok.java:789)
        at lombok.delombok.Delombok.main(Delombok.java:326)
        at lombok.delombok.DelombokApp.runDirectly(DelombokApp.java:176)
        at lombok.delombok.DelombokApp.runApp(DelombokApp.java:47)
        at lombok.core.Main.go(Main.java:130)
        at lombok.core.Main.main(Main.java:43)
```

**Review by ishche — CHANGES_REQUESTED** [review]
lombok.jar is missing, and after manually adding it I'm getting errors too:
```
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/target/generated-sources/antlr4/org/eclipse/lsp/cobol/dialects/daco/DaCoParserBaseVisitor.java:3: error: package org.antlr.v4.runtime.tree does not exist
import org.antlr.v4.runtime.tree.AbstractParseTreeVisitor;
                                ^
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/target/generated-sources/antlr4/org/eclipse/lsp/cobol/dialects/daco/DaCoParserBaseVisitor.java:13: error: cannot find symbol
public class DaCoParserBaseVisitor<T> extends AbstractParseTreeVisitor<T> implements DaCoParserVisitor<T> {
                                              ^
  symbol: class AbstractParseTreeVisitor
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/target/generated-sources/antlr4/org/eclipse/lsp/cobol/dialects/daco/DaCoParserVisitor.java:3: error: package org.antlr.v4.runtime.tree does not exist
import org.antlr.v4.runtime.tree.ParseTreeVisitor;
                                ^
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/target/generated-sources/antlr4/org/eclipse/lsp/cobol/dialects/daco/DaCoParserVisitor.java:12: error: cannot find symbol
public interface DaCoParserVisitor<T> extends ParseTreeVisitor<T> {
                                              ^
  symbol: class ParseTreeVisitor
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/target/generated-sources/antlr4/org/eclipse/lsp/cobol/dialects/daco/DaCoParser.java:4: error: package org.antlr.v4.runtime.dfa does not exist
import org.antlr.v4.runtime.dfa.DFA;
                               ^
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/src/main/java/org/eclipse/lsp/cobol/dialects/daco/MessageServiceParser.java:20: error: package com.google.common.annotations does not exist
import com.google.common.annotations.VisibleForTesting;
                                    ^
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/src/main/java/org/eclipse/lsp/cobol/dialects/daco/MessageServiceParser.java:21: error: package org.antlr.v4.runtime does not exist
import org.antlr.v4.runtime.Parser;
                           ^
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/src/main/java/org/eclipse/lsp/cobol/dialects/daco/MessageServiceParser.java:22: error: package org.antlr.v4.runtime does not exist
import org.antlr.v4.runtime.TokenStream;
                           ^
/Users/ishche/src/che-che4z-lsp-for-cobol/server/dialect-daco/src/main/java/org/eclipse/lsp/cobol/dialects/daco/MessageServiceParser.java:33: error: cannot find symbol
public abstract class MessageServiceParser extends Parser {
                                                   ^
....
```

**Comment by ishche** [comment]
Have you consider to use maven plugin for delomboking?

**Comment by Nurkambay** [comment]
> I am getting errors like
> 
> ```
> ...server\common\src\main\java\org\eclipse\lsp\cobol\common\action\CodeActionProvider.java:19: error: package org.eclipse.lsp4j does not exist
> import org.eclipse.lsp4j.CodeAction;
> ```
> 
> and eventually
> 
> ```
> ...server\common\src\main\java\org\eclipse\lsp\cobol\common\mapping\ExtendedDocument.java:22: error: package org.eclipse.lsp4j does not exist
> import org.eclipse.lsp4j.Position;
>                         ^
> Exception in thread "main" com.sun.tools.javac.processing.AnnotationProcessingError: java.lang.StackOverflowError
>         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:1033)
>         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:937)
>         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1265)
>         at jdk.compiler/com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1380)
>         at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1279)
>         at lombok.delombok.Delombok.delombok(Delombok.java:789)
>         at lombok.delombok.Delombok.main(Delombok.java:326)
>         at lombok.delombok.DelombokApp.runDirectly(DelombokApp.java:176)
>         at lombok.delombok.DelombokApp.runApp(DelombokApp.java:47)
>         at lombok.core.Main.go(Main.java:130)
>         at lombok.core.Main.main(Main.java:43)
> ```

I have checked, this errors do not affect the final delomboked code, it is compilable (if you switch off style check)

**Comment by Nurkambay** [comment]
> Have you consider to use maven plugin for delomboking?

No, I do not see any pluses of it

**Review by slavek-kucera — APPROVED** [review]

**Review by ishche — APPROVED** [review]

**Pull request merged**